care/facility/api/viewsets/asset.py

```diff
@@ -347,7 +347,12 @@ def operate_assets(self, request, *args, **kwargs):
             )
 
         except APIException as e:
-            return Response(e.detail, e.status_code)
+            return Response(
+                {
+                    "detail": f"Communication with the middleware failed.\nReceived status code: {e.status_code}"
+                },
+                status=status.HTTP_424_FAILED_DEPENDENCY,
```

Comment:
502 is a more appropriate for this error as 4xx are usually reserved for client errors
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/502

Reply:
Just a thought, maybe we could test the connection to the asset when updating the configurations, and throw a 4xx error in the update configuration API, that way we are clear it's a misconfiguration.

Reply:
> That'd just not show any error notification, right. Currently we do not handle showing error messages for 5xx status code, afaik.

We could show the error messages for 5xx too right? Not sure why we were not doing that. Was there a reason?

cc: @gigincg

Reply:
related PR : https://github.com/coronasafe/care_fe/pull/6873